### PR TITLE
Bump cyclic dependencies

### PR DESCRIPTION
**Pull request by octogonz**
Upgrade to the latest Heft release which has a bunch of important fixes for Jest

### CODE REVIEW

File: heft-plugins/pre-compile-hardlink-or-copy-plugin/package.json [hunk 1]
```diff
@@ -12,15 +12,15 @@
     "build": "heft test --clean"
   },
   "peerDependencies": {
-    "@rushstack/heft": "^0.4.4"
+    "@rushstack/heft": ">=0.4.4 <1.0.0"
```
Comment (iclanton) [inline heft-plugins/pre-compile-hardlink-or-copy-plugin/package.json:15]: This is isomorphic to what was there before.

File: heft-plugins/pre-compile-hardlink-or-copy-plugin/src/index.ts [hunk 2]
```diff
@@ -79,9 +79,9 @@ export class PreCompileHardlinkOrCopyPlugin implements IHeftPlugin<IPreCompileHa
       options.copyInsteadOfHardlink
     );
     if (options.copyInsteadOfHardlink) {
-      heftConfiguration.terminal.writeLine(`Copied ${linkCount} files`);
+      heftConfiguration.globalTerminal.writeLine(`Copied ${linkCount} files`);
```
Comment (iclanton) [inline heft-plugins/pre-compile-hardlink-or-copy-plugin/src/index.ts:82]: This should request a `ScopedLogger`, not use the `globalTerminal`

File: heft-plugins/pre-compile-hardlink-or-copy-plugin/package.json (hunk 1 above)
Comment (octogonz) [reply heft-plugins/pre-compile-hardlink-or-copy-plugin/package.json:15]: @iclanton That is not true. The `^` and `~` operators work weirdly for versions starting with `0.`

Comment (iclanton) [reply heft-plugins/pre-compile-hardlink-or-copy-plugin/package.json:15]: Huh. TIL.

**Pull request merged**